adobe/spectrum-web-components
# chore(picker):  added a new interaction strategy

Abstracted a new interaction strategy for Picker. This will allow our picker to be stateless and handle all the interactions through a separate `mobile` and `desktop` controllers instead.

## Related issue(s)



- #4227
- #4461
- #4572
- #4459

## Motivation and context



## How has this been tested?




-   [ ] Test 1 (Desktop)
    1. Go [here](https://bug-picker-auto-slection--spectrum-web-components.netlify.app/storybook/?path=/story/picker--default)
    2. click on the picker button
    3. verify that the picker opens
    4. click on the picker button again
    5. verify that the picker is closed
-   [ ] Test 2 (Desktop)
    1. Go [here](https://bug-picker-auto-slection--spectrum-web-components.netlify.app/storybook/?path=/story/picker--default)
    2. longpress on the picker button and make an item selection by dragging the mouse to the menu-item and releasing the pointer
    3. verify the picker opened and actually made a selection before closing.
-   [ ] Test 3 (Mobile)
     1. Go [here](https://bug-picker-auto-slection--spectrum-web-components.netlify.app/storybook/?path=/story/action-menu--default).
     2. Click on the action-menu button and verify a tray opens up.
     3. Select a menu-item from the tray and the tray should close after making a selection.

## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
